### DISCUSSION TITLE
Repeats Start Embedding Every Start of Obsidian

### DISCUSSION THREAD

**Discussion by hittjwX** [brianpetro/obsidian-smart-connections#640] [General]
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/96987960/8901426e-5a31-4daf-a576-d8bbc999e0a0)

Smart Connections repeats the embedding every time I start Obsidian. Right after I open Obsidian, it runs through the vault (22k records) and starts embedding.

I'm using Smart Connections v2.1.67. It does a great job of letting me use Obsidian while embedding, but it takes a while. 

It would be nice if the notification was smaller or didn't cover up the triple dot pull-down menu. Overall, the latest releases are faster.

It would be better if the Smart Connections application console had all these details. Anything that adds time between opening Obsidian and taking notes is a distraction.

**Comment by brianpetro** [comment]
Hey @jwhco 

There was a bug introduced in `2.1.67` that prevented embeddings from saving. Update to the latest version, and it should be fixed!

And thanks for the notes about the UI.

Most notifications can be muted with that little button on the left in your image. That should help the UI issue a bit

🌴

**Reply by hittjwX** [reply]
I'm testing the new version (v2.1.71) right now. It's been running for a long time on 22k notes. However, it does seem to be saving. 

The good news is that I can still create and edit notes. The `Embedding smart_blocks..` doesn't minimize. But I can work around it.

I'm using your `Smart Connect` app (v1.1.55), which significantly increases tokens per second. I get between 2500 and 4000 tokens/sec with and less than 1500 without.

My environment is strictly a local model. BGE-micro-v2. It's not using the GPU, so I want to have a separate server as soon as feasible.

This is pegging out the CPU, but I can still write, check email, and do my typical workload at 60% capacity. 

I have a production mixed-media publishing environment. I've started featuring `Smart Connections` in my editorial workflow videos. Solid implementation here.

**Comment by hittjwX** [comment]
After taking a long time to rebuild due to 22K notes, `Smart Connections` v2.1.71 allows me to restart Obsidian **without** remapping initially. In 3-5 minutes, it starts embedding but finishes very quickly. 

I'm able to work right as Obsidian returns. I'm able to work while the very quick indexing happens again. There were only 25 `smart_blocks` this time, rather than more than 100K.

As I start working, I have delays because it might refresh every 20 minutes or so. When that happens the CPU pegs out 100%. If it used the GPU that would be better. 

This is significantly better. I can live with a brief refresh. Better if it was in the `Smart Connect` app, or only if the system is idle. Or never after I get a note open. If you can think of anything else I should test, please let me know, otherwise this is closed.
